src/main/java/com/aws/greengrass/deployment/DeviceConfiguration.java

```diff
@@ -791,4 +793,39 @@ public KeyManager[] getDeviceIdentityKeyManagers() throws TLSAuthException {
     public Topics getHttpClientOptions() {
         return getTopics("httpClient");
     }
+
+    /**
+     * Set device config based on existing System property.
+     */
+    private void handleExistingSystemProperty() {
+        //handle s3 endpoint type
+        if (System.getProperty(S3_ENDPOINT_PROP_NAME) != null
+                && System.getProperty(S3_ENDPOINT_PROP_NAME).equalsIgnoreCase(S3EndpointType.REGIONAL.name())) {
+            gets3EndpointType().withValue(S3EndpointType.REGIONAL.name());
+        }
+        //handle fips mode
+        String useFipsMode = System.getProperty(SdkSystemSetting.AWS_USE_FIPS_ENDPOINT.property());
+        if (useFipsMode != null && Coerce.toBoolean(useFipsMode)) {
```

Comment:
Nit: `Coerce.toBoolean` can handle null. 

Shouldn't we set the system properties based on the device config?
In L 376, we're setting the system property for fips with the device config. So, why are we updating the device config for fips based on the system property again? [getRegionValidator() comes before handleExistingSystemProperty()](https://github.com/aws-greengrass/aws-greengrass-nucleus/blob/fa6cedb36481062cf4fd8213bd4fe31e5f1d9e63/src/main/java/com/aws/greengrass/deployment/DeviceConfiguration.java#L160-L164)

Reply:
Updated